### PR TITLE
Upgrade to OWL API 4.5.29.

### PR DESCRIPTION
**Pull request by balhoff**
I updated a bunch of tests, but had to ignore a few due to some unfamiliarity and suspected disused code. This OWL API upgrade is important for maintaining OBO format parity with Protege and ROBOT.

### CODE REVIEW

**Comment by balhoff** [comment]
I don't think we have CI checks enabled here, but the remaining tests do all pass for me locally.

File: OWLTools-Core/src/test/java/owltools/mooncat/DiffUtilTest.java
Comment (matentzn) [inline OWLTools-Core/src/test/java/owltools/mooncat/DiffUtilTest.java:1]: Assuming these different are accounted for!

**Review by matentzn — APPROVED** [review]
I mean, I trust your own review much more. It looks ok - you skip a lot of tests, but I am assuming you dont do so without a good reason. 

As I am anyways advocating for fading owltools out, I am fine with these changes which I eyeballed.

File: OWLTools-Core/src/test/java/owltools/mooncat/DiffUtilTest.java
Comment (balhoff) [reply OWLTools-Core/src/test/java/owltools/mooncat/DiffUtilTest.java:1]: My not too deep analysis made me think it is related to the new output of a label for `oboInOwl:id`, like noted here: https://github.com/owlcs/owlapi/pull/1102#issuecomment-2089088246

**Comment by balhoff** [comment]
@kltm if I merge this PR, would it immediately start getting built and used in the GO pipeline, or is the pipeline getting owltools from ODK now?

**Comment by kltm** [comment]
@balhoff Mixed bag, perhaps.
Ontology production uses the ODK (`'obolibrary/odkfull:v1.4'`) in various branches. However, owltools also gets used for building the solr index and doing "predictions" in the pipeline; these seem to follow a local build aimed at `master`.

**Review by kltm — APPROVED** [review]
@balhoff eyeballing the changes, mostly adding ignores and updating endpoints, it seems to make sense. Everything else I would leave to your judgement.

**Comment by balhoff** [comment]
@kltm I was thinking about the timing of things. I would merge this into master and then make an update to ODK. Once that is out the ontology build will start using it. But are you okay with the pipeline immediately using the new OWL API? I think it will be okay for the ontology, since it won't be saving out ontology files.

**Comment by kltm** [comment]
@balhoff Nah--my only ask would be that this is done either late today or early tomorrow, to give the current `snapshot` a chance to finish with a guaranteed version throughout the run.

**Pull request merged**